src/parser/Parser/Parser.hs

```diff
@@ -900,73 +917,89 @@ expression = makeExprParser expr opTable
-                         return (Unary (meta pos) operator)))
+                         emeta' <- opRange emeta
+                         return (Unary emeta' operator)))
       prefix s operator =
-          Prefix (do pos <- getPosition
+          Prefix (do emeta <- meta . newPos <$> getPosition
                      reservedOp s
-                     return (Unary (meta pos) operator))
+                     emeta' <- opRange emeta
+                     return (Unary emeta' operator))
       op s binop =
-          InfixL (do pos <- getPosition
+          InfixL (do emeta <- meta . newPos <$> getPosition
                      withLinebreaks $ reservedOp s
-                     return (Binop (meta pos) binop))
+                     emeta' <- opRange emeta
+                     return (Binop emeta' binop))
 
       arrayAccess =
-          Postfix (do pos <- getPosition
+          Postfix (do emeta <- meta . newPos <$> getPosition
                       index <- parens expression
-                      return (\target -> ArrayAccess{emeta = meta pos
+                      emeta' <- opRange emeta
+                      return (\target -> ArrayAccess{emeta = emeta'
                                                     ,target
                                                     ,index
                                                     }))
 
       consume =
-          Prefix (do pos <- getPosition
+          Prefix (do emeta <- meta . newPos <$> getPosition
                      reserved "consume"
-                     return (Consume (meta pos)))
+                     emeta' <- opRange emeta
+                     return (Consume emeta'))
 
       typedExpression =
-          Postfix (do pos <- getPosition
+          Postfix (do emeta <- meta . newPos <$> getPosition
                       withLinebreaks colon
-                      t <- typ
-                      return (\e -> TypedExpr (meta pos) e t))
+                      ty <- typ
+                      emeta' <- opRange emeta
+                      return (\body -> TypedExpr{emeta = emeta'
+                                                ,body
+                                                ,ty}))
       messageSend =
-          Postfix (do pos <- getPosition
+          Postfix (do emeta <- meta . newPos <$> getPosition
                       withLinebreaks bang
                       name <- Name <$> identifier
                       typeArguments <- option [] (try . brackets $ commaSep typ)
                       args <- parens arguments
-                      let msgSend opt target = MessageSend {emeta = meta pos
-                                                           ,typeArguments
-                                                           ,target
-                                                           ,name
-                                                           ,args}
-                      return $ msgSend False)
+                      emeta' <- opRange emeta
```

Comment:
wouldn't it make sense to abstract the two lines:
```haskell
do emeta <- meta . newPos <$> getPosition
     emeta' <- opRange emeta
```

into a single function that returns the range?